src/extractor/guidance/intersection_handler.cpp

```diff
@@ -382,19 +386,17 @@ IntersectionHandler::getNextIntersection(const NodeID at, const EdgeID via) cons
     // Starting at node `a` via edge `e0` the intersection generator returns the intersection at `c`
     // writing `tl` (traffic signal) node and the edge `e1` which has the intersection as target.
 
-    NodeID node = SPECIAL_NODEID;
-    EdgeID edge = SPECIAL_EDGEID;
-
-    std::tie(node, edge) = intersection_generator.SkipDegreeTwoNodes(at, via);
-    auto intersection = intersection_generator(node, edge);
-
+    const auto intersection_parameters = intersection_generator.SkipDegreeTwoNodes(at, via);
     // This should never happen, guard against nevertheless
-    if (node == SPECIAL_NODEID || edge == SPECIAL_EDGEID)
+    if (intersection_parameters.nid == SPECIAL_NODEID ||
+        intersection_parameters.via_eid == SPECIAL_EDGEID)
```

Comment:
It shouldn't. It could be, that we change `SkipDegreeTwo` to return invalid IDs, in case we run into a dead end in the graph. So I feel it is safer to not assume too much about what happens when finding the next intersection